### ISSUE TITLE
🔗 📊 🐞 Bug Report: Menu widget causing significant slow down in editor (Duplicate of #21956) - [ED-10424]

### ISSUE THREAD

**Issue by WebAgencyBlink** [elementor/elementor#23715]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest stable version of Elementor.

### Description

I am using the Menu widget in my Theme Builder > Header and experiencing performance issues. Whenever I go to edit the fields in the Menu Items on the left hand side, the browser slows down and seems to hang. This happens even for simple stuff like changing a menu items title. If you keep editing it gets to the point where it infinitely hangs and you can't even save your work.

At one point I had the chrome alert come up asking if i want to exit the process or wait because it was hanging for so long. I am not doing anything special besides trying to edit the menu item fields.

### Steps to reproduce

1. Create a header in your theme builder
2. Add a menu widget to the header
3. Fill a couple menu items with some containers and layouts
4. Edit the menu item fields

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt



					
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.24.0
	MySQL version: mariadb.org binary distribution v10.11.5-MariaDB-1:10.11.5+maria~ubu2004
	PHP Version: 8.0.30
	PHP Memory Limit: 256M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3.1
	Site URL: https://stg-mcim24x7-staging.kinsta.cloud
	Home URL: https://stg-mcim24x7-staging.kinsta.cloud
	WP Multisite: No
	Max Upload Size: 128 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /resources/%postname%/
	Language: en-US
	Timezone: -4
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.8.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields PRO
		Version: 6.2.0
		Author: WP Engine

	Duplicate Page
		Version: 4.5.2
		Author: mndpsingh287

	DynamicConditions
		Version: 1.6.2
		Author: RTO GmbH

	Elementor
		Version: 3.15.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.15.1
		Author: Elementor.com

	HubSpot All-In-One Marketing - Forms, Popups, Live Chat
		Version: 10.2.3
		Author: HubSpot

	Lazy Load Control For Elementor
		Version: 0.0.2
		Author: Jose Mortellaro

	Safe SVG
		Version: 2.2.0
		Author: 10up

	Usersnap
		Version: 4.18
		Author: Usersnap


== Elements Usage ==
	
	archive : 2
		container : 29
		form : 2
		heading : 2
		loop-grid : 4
		shortcode : 1
		spacer : 5
		text-editor : 4
	container : 1
		container : 1
		heading : 1
		loop-grid : 1
		shortcode : 1
		text-editor : 1
	footer : 1
		container : 12
		form : 2
		heading : 3
		nav-menu : 1
		social-icons : 1
		text-editor : 8
		theme-site-logo : 1
	header : 1
		button : 9
		container : 28
		heading : 7
		icon : 1
		image : 3
		loop-grid : 1
		mega-menu : 1
		nav-menu : 5
		text-editor : 4
		theme-site-logo : 2
	section : 2
		button : 1
		container : 3
		counter : 1
		heading : 1
		image : 1
		text-editor : 2
	widget : 0
		button : 2
	wp-page : 0
		animated-headline : 1
		button : 3
		container : 48
		counter : 3
		heading : 11
		image : 10
		nested-tabs : 1
		shortcode : 3
		text-editor : 6
	single-post : 7
		container : 127
		form : 1
		heading : 26
		image : 8
		loop-grid : 1
		shortcode : 14
		spacer : 6
		text-editor : 15
		theme-post-content : 2
		video : 1
	single-page : 4
		animated-headline : 2
		button : 10
		container : 127
		counter : 3
		form : 1
		heading : 22
		image : 11
		nested-tabs : 1
		shortcode : 10
		spacer : 7
		text-editor : 13
	loop-item : 22
		container : 92
		counter : 1
		heading : 22
		icon : 6
		image : 23
		post-info : 2
		text-editor : 19
		theme-post-excerpt : 1
		theme-post-title : 1


== Settings ==
	
	cpt_support: post, page, e-landing-page, case_study, platform_module, solution
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap


== Features ==
	Custom Fonts: 1
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active by default
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	Lazy Load Control For Elementor: Incompatible

== Elementor Pro - Compatibility Tag ==
	
	Lazy Load Control For Elementor: Incompatible
				

````

</details>

**Comment by nicholaszein** [comment]
Hello @WebAgencyBlink!

Thank you for your submission! 🙏 

### 🔗 🐞 This issue is a Duplicate of #21956 

Which is on the same subject. 

For this reason, I'm closing this thread and 🔃 **Merging** it with the original, so we can have all the discussions in one location, and deal with it more efficiently.

Feel free to continue the conversation there. 😉 

### 🔍 Please, in the future, remember to check if there isn't any thread similar to yours already opened. 

Kind regards

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved with the release of Elementor v3.23.0! 🥳 

The time to interaction should improve by 60%-98%, and performance overall should be better. For a full breakdown please read [this article](https://developers.elementor.com/elementor-3-23-developers-update/#:~:text=Performant%20Nested%20Elements%20In%20the%20Editor).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
